grails-app/controllers/au/org/ala/merit/OrganisationController.groovy

```diff
@@ -450,20 +459,87 @@ class OrganisationController {
 
     }
 
-    def viewOrganisationReport(String reportId) {
-        viewOrEditOrganisationReport(reportId, false)
+
+    def viewOrganisationReport(OrganisationReportCommand cmd) {
+        if (cmd.hasErrors()) {
+            error(cmd.errors.toString())
+            return
+        }
+
+        if (cmd.report.type == ReportService.PERFORMANCE_MANAGEMENT_REPORT) {
+            viewOrEditOrganisationReport(cmd.report, false)
+        }
+        else {
+            render model:cmd.model, view:'/activity/activityReportView'
+        }
+    }
+
+
+    def editReport(String id, String reportId) {
+        if (!id || !reportId) {
+            error('An invalid report was selected for data entry', id)
+            return
+        }
+
+        Map model = activityReportModel(id, reportId, ReportMode.EDIT, params.getInt('formVersion', null))
+
+        if (!model.editable) {
+            redirect action:'viewReport', id:id, params:[reportId:reportId, attemptedEdit:true]
+        }
+        else {
+            if (model.config.requiresActivityLocking) {
+                Map result = reportService.lockForEditing(model.report)
+                model.locked = true
+            }
+            model.saveReportUrl = createLink(controller:'organisation', action:'saveReport', id:id, params:[reportId:reportId])
+            render model:model, view:'/activity/activityReport'
+        }
     }
 
-    def editOrganisationReport(String reportId) {
-        viewOrEditOrganisationReport(reportId, true)
+
+    def saveReport(SaveReportDataCommand saveReportDataCommand) {
+        Map result
+        if (saveReportDataCommand.report?.organisationId != params.id) {
+            result = [status:HttpStatus.SC_UNAUTHORIZED, error:"You do not have permission to save this report: check if the report belongs to this management unit: " + params?.id ]
+        }
+        else {
+            result = saveReportDataCommand.save()
+        }
+
+        render result as JSON
+
     }
 
-    private def viewOrEditOrganisationReport(String reportId, Boolean edit) {
-        Map report = reportService.get(reportId)
+    @PreAuthorise(accessLevel = 'admin')
+    def resetReport(String id, String reportId) {
+        if (!id || !reportId) {
+            error('An invalid report was selected', id)
+            return
+        }
+        Map result = reportService.reset(reportId)
+        render result as JSON
+    }
+
+
+    def editOrganisationReport(OrganisationReportCommand cmd) {
```

Comment:
Missing authorization annotation

Reply:
has been deleted as the method being used is editReport